src/Components/ContextMenu.re

```diff
@@ -0,0 +1,257 @@
+          // iconView
+           labelView </View>
+      </Clickable>;
+      // onMouseOut={_ => setIsFocused(_ => false)}
+      // onMouseOver={_ => setIsFocused(_ => true)}
+    };
+};
+
+// MENU
+
+module Menu = {
+  module Styles = {
+    open Style;
+
+    let container = (~x, ~y, ~theme: Theme.t) => [
+      position(`Absolute),
+      top(y),
+      left(x),
+      backgroundColor(theme.menuBackground),
+      color(theme.menuForeground),
+      width(Constants.menuWidth),
+    ];
+  };
+
+  let component = React.Expert.component("Menu");
+  let make = (~model, ~theme, ~font, ~onItemSelect, ()) =>
+    component(hooks => {
+      let ((maybeRef, setRef), hooks) = Hooks.state(None, hooks);
+      let {x, y, originX, originY, items, _} = model;
+
+      let height =
+        switch (maybeRef) {
+        | Some((node: node)) => node#measurements().height
+        | None => List.length(model.items) * 20
+        };
+      let width = Constants.menuWidth;
+
+      let x =
+        switch (originX) {
+        | `Left => x
+        | `Middle => x - width / 2
+        | `Right => x - width
+        };
+
+      let y =
+        switch (originY) {
+        | `Top => y - height
+        | `Middle => y - height / 2
+        | `Bottom => y
+        };
+
+      (
+        // TODO: BoxShadow apparently blocks mouse events. Figure out why before adding back
+        // <BoxShadow
+        //   boxShadow={Style.BoxShadow.make(
+        //     ~xOffset=-11.,
+        //     ~yOffset=-11.,
+        //     ~blurRadius=25.,
+        //     ~spreadRadius=0.,
+        //     ~color=Color.rgba(0., 0., 0., 0.2),
+        //     (),
+        //   )}>
+        <View
+          style={Styles.container(~x, ~y, ~theme)}
+          ref={node => setRef(_ => Some(node))}>
+          {items
+           |> List.map(item => {
+                let onClick = () => onItemSelect(item);
+                <MenuItem item theme font onClick />;
+              })
+           |> React.listToElement}
+        </View>,
+        // </BoxShadow>,
+        hooks,
+      );
+    });
+};
+
+// OVERLAY
+
+module Overlay = {
+  module Styles = {
+    open Style;
+
+    let overlay = [
+      position(`Absolute),
+      top(0),
+      bottom(0),
+      left(0),
+      right(0),
+      pointerEvents(`Allow),
+    ];
+  };
+
+  let make = (~model, ~theme, ~font, ~onOverlayClick, ~onItemSelect, ()) => {
+    <Clickable onClick=onOverlayClick style=Styles.overlay>
+      <Menu model theme font onItemSelect />
+    </Clickable>;
+  };
+};
+
+module Make = (()) => {
+  let id = Id.create();
+
+  let init = items => {
+    id,
+    x: 0,
+    y: 0,
+    originX: `Left,
+    originY: `Bottom,
+    items,
+  };
+
+  module Anchor = {
+    let component = React.Expert.component("Anchor");
+    let make =
+        (
+          ~model as maybeModel,
+          ~originX=`Left,
+          ~originY=`Bottom,
+          ~onUpdate,
+          (),
+        ) =>
```

Comment:
We could use mutation and internal bookkeeping instead of passing in `model` and `onUpdate` here, but then there wouldn't be a state update to trigger a re-render.